Added recipe for taproom

dotsdl opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
or on our [Keybase chat](https://keybase.io/team/condaforge.chat)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/taproom```) and found it was in an excellent condition.

dotsdl commented:
@slochower, do you agree to be a recipe maintainer?

dotsdl commented:
@conda-forge-admin, please restart ci

dotsdl commented:
Persistent failures of OSX build after two attempts. We see:

```
Configuring conda.
+ setup_conda_rc ./ ./recipes ./.ci_support/osx64.yaml
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/setup_conda_rc", line 7, in <module>
    from conda_forge_ci_setup.build_utils import setup_conda_rc
ModuleNotFoundError: No module named 'conda_forge_ci_setup'
```

@conda-forge/staged-recipes is this a known issue?

synapticarbors commented:
@dotsdl - the osx failure is a known issue. There is currently no fix, but hopefully it will be resolved soon

dotsdl commented:
Ah, thank you @synapticarbors! Should we wait for resolution, or is this not a blocker for merge? This is a pure-python package, so no OSX-dependent build components.

synapticarbors commented on recipes/taproom/meta.yaml:
```diff
@@ -0,0 +1,41 @@
+    - python
+    - pip
+  run:
+    - python
```
I think you'll need `setuptools` as a `run` dependency since the entry points use `pkg_resources`. This isn't obvious from the import test of `taproom` since it doesn't actually hit the code that uses this.

synapticarbors commented:
I think since this recipe is `noarch: python` it only really needs to build on linux. I'd be more concerned about seeing osx pass if it was complicated with a bunch of dependencies.

dotsdl commented:
Awesome, thank you @synapticarbors! Since we are choosing not to run the tests for this package (this is really just a data package for e.g. [pAPRika](https://github.com/slochower/pAPRika)), I've removed the requirements for testing. I've also added `setuptools` to the run requirements on your suggestion; thank you for that catch!

synapticarbors commented:
I'm happy to merge this once @slochower agrees to be the maintainer.

dotsdl commented:
Added the following maintainers to the recipe:
- @dotsdl 
- @slochower 
- @jeff231li

dotsdl commented:
I agree to be a maintainer, fwiw. :grin:

jeff231li commented:
> Added the following maintainers to the recipe:
> 
> * @dotsdl
> * @slochower
> * @jeff231li

I agree to be a maintainer.

slochower commented:
I agree to be maintainer.

dotsdl commented:
Thanks again for your review and assistance @synapticarbors!
